src/app/routes/api/v3/forms/__tests__/public-forms.verification.routes.spec.ts

```diff
@@ -231,4 +235,249 @@ describe('public-forms.verification.routes', () => {
       expect(response.body).toEqual(expectedResponse)
     })
   })
+
+  describe('POST /forms/:formId/fieldverifications/:transactionId/fields/:fieldId/otp/verify', () => {
+    beforeAll(() => {
+      jest.spyOn(bcrypt, 'compare').mockResolvedValue(true)
```

Comment:
does this need to be mocked? we should try to only mock things we cannot simulate in integration tests; we should instead  call the generate end point for real and see if verify returns success.

see our [auth integration tests](https://github.com/opengovsg/FormSG/blob/develop/src/app/routes/api/v3/auth/__tests__/auth.routes.spec.ts#L451)

Reply:
good point! will do! will wait for otp generation to be merged first because this change is dependent on that